### PR TITLE
Add documentation about warnings

### PR DESCRIPTION
**Pull request by davetcoleman**
See discussion https://github.com/catkin/catkin_tools/issues/199#issuecomment-94541972

### CODE REVIEW

File: docs/verbs/catkin_build.rst [hunk 1]
```diff
@@ -533,6 +533,15 @@ So if you want to check for failing tests, use the ``catkin_test_results`` comma
...
 The result code will be non-zero unless all tests passed.
 
+Building With Warnings
+^^^^^^^^^^^^^^^^^^^^^^
+
+It can sometimes be useful to compile with strict warnings enabled across your whole catkin workspace. To achieve this, use a command like:
+
+.. code-block:: bash
+
+    $ catkin build -v --cmake-args -DEXTRA_C_FLAGS="-Wall -g -W -Wno-unused-parameter"
```
Comment (wjwwood) [inline docs/verbs/catkin_build.rst]: I don't think I've ever seen `EXTRA_C_FLAGS`, is that a CMake thing? I've always used `CMAKE_C_FLAGS` or `CMAKE_CXX_FLAGS`, do you know if `EXTRA_C_FLAGS` is documented anywhere?

Comment (wjwwood) [inline docs/verbs/catkin_build.rst]: `cmake` complains at me when I use `EXTRA_C_FLAGS` in an empty project, but not when I use `CMAKE_C_FLAGS`:

```
william@farl /tmp/test_extra_c_flags
% cat CMakeLists.txt
cmake_minimum_required (VERSION 2.8.11)
project (test)

william@farl /tmp/test_extra_c_flags
% mkdir build

william@farl /tmp/test_extra_c_flags
% cd build

william@farl /tmp/test_extra_c_flags/build
% cmake .. -DEXTRA_C_FLAGS=-Wall
-- The C compiler identification is AppleClang 6.1.0.6020049
-- The CXX compiler identification is AppleClang 6.1.0.6020049
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    EXTRA_C_FLAGS


-- Build files have been written to: /tmp/test_extra_c_flags/build

william@farl /tmp/test_extra_c_flags/build
% rm -rf ./*
zsh: sure you want to delete all the files in /tmp/test_extra_c_flags/build/. [yn]? y

william@farl /tmp/test_extra_c_flags/build
% cmake .. -DCMAKE_C_FLAGS=-Wall
-- The C compiler identification is AppleClang 6.1.0.6020049
-- The CXX compiler identification is AppleClang 6.1.0.6020049
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/test_extra_c_flags/build
```

You might want to check your usage. The packages you are using may process the `EXTRA_C_FLAGS` option, but the documentation should probably not a assume that.

I'll merge this and change it to `CMAKE_C_FLAGS`, and if I'm wrong we can touch it up afterwards.

**Comment by davetcoleman** [comment]
I got that from @jbohren a long time ago, when I was asking him how to do this. I'll defer to him. But this command did work for me today.

File: docs/verbs/catkin_build.rst (hunk 1 above)
Comment (wjwwood) [inline docs/verbs/catkin_build.rst]: One other thing I changed was that you shouldn't include `-g` in the flags, but instead you should set the `CMAKE_BUILD_TYPE` to `Debug` or similar.

**Comment by wjwwood** [comment]
Hmm, that's strange, I'd be interested to see if it is an undocumented feature of CMake or something else.

**Comment by wjwwood** [comment]
Either way it's in there now, if @jbohren comes back with a reason to use `EXTRA_C_FLAGS`, then I'll change the docs again.

I merged with 13c7898. Thanks!

**Pull request closed without merging**